scripts/generate_serialization.py

```diff
@@ -529,6 +537,28 @@ def generate_class_code(class_entry):
                     raise Exception(f"Duplicate enum entry \"{enum_entry}\"")
                 base_class_object.children[enum_entry] = new_class
 
+    # Ensure that there are no duplicate names in the inheritance tree
+    for base_class in base_classes:
+        if base_class.base is None:
+            # Root base class, now traverse the children
+            def traverse_children(node, parents: list, seen: set):
```

Comment:
Move this function out of the loop so it doesn't get defined (again) on each iteration?